plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+        Safe to run outside of launch.
+        """
+        # Process any missed messages, and schedule prompt or delivery for the current ticket as needed
+        if self.current_ticket:
+            logger.debug("Loading moderator {}.".format(self.modid))
+            if self.current_ticket.stage == TicketStage.NEW:
+                # In this case, the ticket at the top of their queue wasn't delivered.
+                # We ignore missed messages since we can't be certain what ticket they refer to
+
+                # Schedule ticket delivery
+                await self.deliver()
+            else:
+                # The last ticket was delivered, but not yet commented
+                # Replay any messages we missed, and process the first message as a comment, if it exists
+                # We can't tell what the rest of the messages refer to, so we ignore them
+
+                # Message snowflake to process from
+                last_read = discord.Object(max(self.last_read_msgid or 0, self.current_ticket.delivered_id))
+
+                # Collect the missed messages
+                mod_messages = []
+                if self.user:
+                    messages = await self.user.history(after=last_read, limit=None).flatten()
+                    mod_messages = [message for message in messages if message.author.id == self.modid]
+
+                if mod_messages:
+                    logger.debug("Missed {} messages from moderator {}.".format(len(mod_messages), self.modid))
+
+                    # Process the first missed message
+                    await self.process_message(mod_messages[0])
+                    # Save the last missed message as the last one handled
+                    if len(mod_messages) > 1:
+                        self.last_read_msgid = mod_messages[-1].id
+                else:
+                    # If we didn't process a comment, schedule the reminder prompt for the current ticket
+                    await self.schedule_prompt()
+
+    def unload(self):
+        """
+        Unload the TicketMod.
+        """
+        self.cancel()
+
+    def cancel(self):
+        """
+        Cancel TicketMod scheduled tasks.
+        """
+        if self._prompt_task and not self._prompt_task.cancelled() and not self._prompt_task.done():
+            self._prompt_task.cancel()
+        if self._delivery_task and not self._delivery_task.cancelled() and not self._delivery_task.done():
+            self._delivery_task.cancel()
+
+    def get_current_ticket(self) -> Ticket:
+        # Get current ticket
+        ticket = fetch_tickets_where(
+            modid=self.modid,
+            stage=[TicketStage.DELIVERED, TicketStage.NEW],
+            _extra="ORDER BY stage DESC, id ASC LIMIT 1"
+        )
+        return next(ticket, None)
+
+    def perm_check(self):
+        """
+        Check that this user has moderator permissions.
+        """
+        # TODO
+        return True
```

Comment:
Fairly safe to ignore, provided we can handle the resulting HTTP errors gracefully.